File: Symulator Świata/dandelion.py
import random
from plant import Plant
GROW_CHANCE = 30


class Dandelion(Plant):

    def __init__(self, world, coord=None):
        self.name = 'd'
        self._strength = 0
        self._initiative = 0
        self._age = 0
        self.sprite = (235, 235, 145)
        self.world = world
        if coord is not None:
            self.world.board[coord] = self
        else:
            self.spawn()

    def action(self, coord):
        i = 0
        while i < 3:
            i += 1
            growRoll = random.randint(0, 100)
            if growRoll <= GROW_CHANCE:
                self.multiply(coord)

    def createChild(self, coord):
        d1 = Dandelion(self.world, coord)

File: PO PROJEKT 1/PO PROJEKT 1/PROJEKT PO PLIKI ŹRÓDŁOWE/Dandelion.cpp
#include "Dandelion.h"



Dandelion::Dandelion(World* world) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'd';
	this->world = world;
	spawn();
}

Dandelion::Dandelion(World* world, Coordinate coord) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'd';
	this->world = world;
	world->board.insert({ coord, this });
}

Organism* Dandelion::createChild(Coordinate coord) {
	return new Dandelion(world, coord);
}

void Dandelion::action(Coordinate coord) {
	int growRoll = rand() % 100;
	int growTimes = 0;
	if (growRoll <= GROW_CHANCE) {
		while (growTimes < 3) {
			multiply(coord);
			growTimes++;
		}
	}
}

void Dandelion::draw() {
	cout << BRIGHT_YELLOW << getSprite() << RESET;
}
